### PR TITLE
Add info to docs about working with github enterprise server

### PR DESCRIPTION
**Pull request by capnspacehook**
Document added field in the GitHub Authentication Connector added in #17293 and https://github.com/gravitational/teleport.e/pull/527.

### CODE REVIEW

File: docs/pages/access-controls/sso/github-sso.mdx [hunk 1]
```diff
@@ -40,6 +41,8 @@ Instructions for creating a GitHub OAuth app are available here:
 Define a GitHub authentication connector by creating a file called `github.yaml`
 with the following content:
 
+<ScopedBlock scope="oss">
```
Comment (alexfornuto) [inline docs/pages/access-controls/sso/github-sso.mdx:44]: We prefer to use Tabs in cases like this for readers who forgot to (or don't realize they can) select their environment at the top of the page.

I'll submit a commit to adjust.

Comment (alexfornuto) [reply docs/pages/access-controls/sso/github-sso.mdx:44]: Also, should the Enterprise version also apply to Cloud?

**Review by alexfornuto — COMMENTED** [review]
As specified in one of my review comments I have changes incoming. Waiting to hear back from the author if they prefer commits to this branch or a PR to it. Included in those changes are:
- Updating `config.json` to allow scopes on this page.
- Replacing `ScopedBlock` with `TabItem` for different config examples, for readers who forgot to change their scope at the top of the page.

EDIT: Staged as #18755

File: docs/pages/access-controls/sso/github-sso.mdx [hunk 2]
```diff
@@ -11,7 +11,8 @@ Teleport.
 ## Prerequisites
 
 - A GitHub organization with at least one team. <ScopedBlock scope="oss">This organization must not have external SSO set up, or Teleport
-will refuse to create the GitHub authentication connector.</ScopedBlock>
+will refuse to create the GitHub authentication connector.</ScopedBlock><ScopedBlock scope="enterprise">This organization can be hosted
+from either GitHub Cloud or Github Enterprise Server.</ScopedBlock>
```
[the remark this answers is not in the document]
Comment (capnspacehook) [reply docs/pages/access-controls/sso/github-sso.mdx:15]: Ah yep it does, good catch

**Review by alexfornuto — APPROVED** [review]

**Review by jakule — APPROVED** [review]

**Pull request merged**